docs/Contributing.md

```diff
@@ -25,44 +25,43 @@ To contribute, please fork the repo and develop in a feature branch off the curr
+
 * We use explicit types (no usages of `var`)
-* We always specify the visibility, even if it's the default (i.e. private string _foo not string _foo). Visibility should be the first modifier (i.e. public abstract not abstract public).
-* We avoid this. unless absolutely necessary.
-* We use `_camelCase` for private fields and use readonly where possible. When used on static fields, readonly should come after static (i.e. static readonly not readonly static).
-* Namespace imports should be specified at the top of the file, outside of namespace declarations and should be sorted alphabetically with the exception of `System.*` namespaces, which are placed first.
-* We use language keywords instead of BCL types (i.e. `int`, `string`, `float` instead of `Int32`, `String`, `Single`, etc) for both type references as well as method calls (i.e. `int.Parse` instead of `Int32.Parse`).
+* We avoid this, unless absolutely necessary.
+* We use `_camelCase` for private fields.
+* Use readonly where possible.
 * We use PascalCasing to name all our methods, properties, constant local variables, and static readonly fields.
 * We use `nameof(...)` instead of `"..."` whenever possible and relevant.
-* Fields should be specified at the top within type declarations.
+* We use [nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references) to make conscious decisions on the nullability of references, be more clear with our intent and reduce `NullReferenceException`s. Add `#nullable enabled` to the top of all the modified files unless:
+  * Nullable reference types are already enabled for the file
+  * The file is in one of the test projects
+  * The changes you are introducing to the file are negligable in size compared to the size of the whole file,
+  * You don't have enough context on the code to make decisions on nullability of types.
+
+Some of the analyzer rules are currently being treated as "info/suggestion"s instead of "warning"s, because we have not yet done a solution wide refactoring to comply with the rules. Although it would be most welcome, you are not required to fix any of the existing suggestions. However, any code that you introduce should be free of suggestions.
 
 [Top](#top)
 
 # Branching information #
 
+We do development in *master* branch. After a release branch is created, any new changes that should be included in that release are cherry-picked from *master*.
+
+We follow the same versioning as https://github.com/dotnet/sdk and release branches are named after the version numbers. For instance, `release/5.0.2xx` branch ships with .Net SDK 5.0.102.
+
 | Topic | Branch |
 |-------|-------|
-| Development | *stabilize* |
-| Next | *master* |
-| Current Release | *release/2.1* |
+| Development | *master* |
+| Current Release | *release/6.0.1xx-preview1* |
```

Comment:
I'm not fan of this, this will get outdated again, lets just mention releases are in `release/*`, because current release is not *release/6.0.1xx-preview1* its in *release/5.0.2xx*...

Reply:
Removed the table.